### PR TITLE
`announcement` - clickable link

### LINKED ISSUE #1502: `announcements` - clickable links
#### What would you like to be added:
For our announcements banners, make it "clickable" when there's a URL present on the announcement item.  Add a caret when the banner is clickable
#### Why is this needed:
If we put a message that, "an update is needed", it should be able to redirect to app stores.

### PR DESCRIPTION
**Pull request by kyleleow**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
- To allow banner to navigate to external url, e.g.: app store link for latest version, webpage for guide

#### Which issue(s) does this PR fixes?:

Fixes #1502

#### Additional comments?:
- Change to get version from `package.json`, we should use version from `package.json` moving forward, since `nativeApplicationVersion` is always `null` for `web`

#### Developer Checklist:

- [x] Read your code changes at least once
- [x] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [x] No console errors on web
- [x] Tested on Light mode and Dark mode*
- [ ] Your UI implementation visually matched the rendered design*
- [ ] Unit tests*
- [ ] Added e2e tests*
- [ ] Added translations*

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -18,7 +18,8 @@ module.exports = function (api) {
           '@screens': './mobile-app/app/screens',
           '@store': './shared/store',
           '@translations': './shared/translations',
-          '@tailwind': './mobile-app/app/tailwind'
+          '@tailwind': './mobile-app/app/tailwind',
+          '@package': './package.json'
```
Comment (thedoublejay) [inline babel.config.js:22]: need to remove this

File: package.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
   "name": "@defichain/wallet",
+  "version": "0.17.0",
```
Comment (thedoublejay) [inline package.json:3]: same here

File: mobile-app/app/screens/AppNavigator/screens/Balances/components/Announcements.tsx [hunk 3]
```diff
@@ -15,33 +18,79 @@ export function Announcements (): JSX.Element {
...
+        testID='announcements_banner'
+        style={tailwind('px-4 py-3 flex-row items-center')} light={tailwind('bg-warning-50')}
+        dark={tailwind('bg-darkwarning-50')}
+        onPress={async () => await openURL(announcement.url)}
+      >
+        <ThemedIcon style={tailwind('mr-2')} iconType='MaterialIcons' name='campaign' size={22} />
+        <ThemedText style={tailwind('text-xs flex-auto')} testID='announcements_text'>
+          {`${announcement.content} `}
+          <View>
+            <ThemedIcon
```
Comment (thedoublejay) [inline mobile-app/app/screens/AppNavigator/screens/Balances/components/Announcements.tsx:40]: Can we make this fixed on the right side? Not at the end of the content but as a fixed button on the right (if link is existing)

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: mobile-app/app/screens/AppNavigator/screens/Balances/components/Announcements.tsx [hunk 4]
```diff
@@ -15,33 +18,79 @@ export function Announcements (): JSX.Element {
...
+            />
+          </View>
+        </ThemedText>
+      </ThemedTouchableOpacity>
+    )
   }
-  return <></>
+
+  return (
+    <ThemedView
```
Comment (thedoublejay) [inline mobile-app/app/screens/AppNavigator/screens/Balances/components/Announcements.tsx:55]: Probably we can make it as same component, to avoid having duplicate per content. Just disable the touchable opacity if there's no link

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: babel.config.js (hunk 1 above)
Comment (kyleleow) [reply babel.config.js]: updated

File: package.json (hunk 2 above)
Comment (kyleleow) [reply package.json]: updated

File: mobile-app/app/screens/AppNavigator/screens/Balances/components/Announcements.tsx (hunk 3 above)
Comment (kyleleow) [reply mobile-app/app/screens/AppNavigator/screens/Balances/components/Announcements.tsx]: moved to right and increased to size `24`, but still pending final design from jonathan

File: mobile-app/app/screens/AppNavigator/screens/Balances/components/Announcements.tsx (hunk 4 above)
Comment (kyleleow) [reply mobile-app/app/screens/AppNavigator/screens/Balances/components/Announcements.tsx]: updated to just use `<ThemedTouchableOpacity>`

File: shared/types/website.ts [hunk 5]
```diff
@@ -13,6 +13,13 @@ export interface AnnouncementData {
    * Versioned matching represented as semver satisfies
    */
   version: string
+  url: {
```
Comment (thedoublejay) [inline shared/types/website.ts:16]: Probably better to make this optional

File: website/src/pages/api/v0/announcements.ts [hunk 6]
```diff
@@ -16,14 +16,28 @@ export default async function handle (req: NextApiRequest, res: NextApiResponse<
       'zh-Hans': '请更新至最新版本应用程式使用去中心化贷款。',
       'zh-Hant': '請更新至最新版本應用程式使用去中心化貸款。'
     },
-    version: '<0.17.0'
+    version: '<0.17.0',
```
Comment (thedoublejay) [inline website/src/pages/api/v0/announcements.ts:19]: Can you revert this change? Even with older users, they can't still update because code for new banner is not yet existing

File: website/src/pages/api/v0/announcements.ts [hunk 7]
```diff
@@ -16,14 +16,28 @@ export default async function handle (req: NextApiRequest, res: NextApiResponse<
...
+    }
   }, {
     lang: {
       en: 'Decentralized loan is now available on the Light Wallet app.',
       de: 'Dezentrale Darlehen sind jetzt in der Light Wallet App verfügbar.',
       'zh-Hans': '去中心化贷款已在轻钱包 Light Wallet 正式推出',
       'zh-Hant': '去中心化貸款已在輕錢包 Light Wallet 正式推出'
     },
-    version: '>=0.17.0'
+    version: '>=0.17.0',
```
Comment (thedoublejay) [inline website/src/pages/api/v0/announcements.ts:34]: Can be removed once optional is set

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: website/src/pages/api/v0/announcements.ts (hunk 6 above)
Comment (kyleleow) [reply website/src/pages/api/v0/announcements.ts:19]: then instead of removing it entirely, why not we target after the version we added announcement, until the one version before the latest version? 

I check that 0.12 is when we added announcement banner

File: website/src/pages/api/v0/announcements.ts (hunk 7 above)
Comment (kyleleow) [reply website/src/pages/api/v0/announcements.ts]: updated

File: shared/types/website.ts (hunk 5 above)
Comment (kyleleow) [reply shared/types/website.ts]: updated

**Review by thedoublejay — APPROVED** [review]
